File: Leetcode/1221. Split a String in Balanced Strings/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var balancedStringSplit = function(s) {
    let count = 0;
    let res = 0;

    for (let i = 0; i < s.length; i++) {
        count += s[i] == "R" ? 1 : -1;
        if (count == 0) {
            res++;
        }
    }

    return res;
};

File: Leetcode/1221. Split a String in Balanced Strings/solution.c
int balancedStringSplit(char * s) {
    int count = 0;
    int res = 0;

    while (*s != '\0') {
        if (*s == 'R') {
            count += 1;
        }
        else {
            count -= 1;
        }

        if (!count) {
            res += 1;
        }
        s += 1;
    }

    return res;
}